build.gradle

```diff
@@ -30,9 +54,62 @@ task javadocJar(type: Jar, dependsOn: javadoc) {
     from javadoc.destinationDir
 }
 
+// add javadoc/source jar tasks as artifacts
+artifacts {
+    archives sourcesJar, javadocJar
+}
 
-dependencies {
-    compile 'com.squareup.okhttp3:okhttp:3.2.0'
-    testCompile "org.mockito:mockito-core:1.+"
-    testCompile group: 'junit', name: 'junit', version: '4.11'
+signing {
+    sign configurations.archives
+}
+
+idea {
+    module {
+        downloadJavadoc = true
+        downloadSources = true
+    }
 }
+
+uploadArchives {
+    repositories {
+        mavenDeployer {
+            beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
+
+            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
+                authentication(userName: ossrhUsername, password: ossrhPassword)
+            }
+
+            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
+                authentication(userName: ossrhUsername, password: ossrhPassword)
+            }
+
+            pom.project {
+                name 'okhttp-eventsource'
+                packaging 'jar'
+                description 'EventSource Implementation built on OkHttp'
+                url 'https://github.com/launchdarkly/okhttp-eventsource'
+
+                scm {
+                    connection 'scm:git:https://github.com/launchdarkly/okhttp-eventsource'
+                    developerConnection 'scm:git:https://github.com/launchdarkly/okhttp-eventsource'
+                    url 'https://github.com/launchdarkly/okhttp-eventsource'
+                }
+
+                licenses {
+                    license {
+                        name 'The Apache License, Version 2.0'
+                        url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                    }
+                }
+
+                developers {
+                    developer {
+                        id 'jkodumal'
+                        name 'John Kodumal'
+                        email 'john@catamorphic.com'
```

Comment:
We can just turn this into a generic "LaunchDarkly Team" unless you feel strongly about listing individual contributors. In either case let's use launchdarkly.com instead of catamorphic.